services/ui-src/src/components/overlays/EntityDetailsMultiformOverlay.tsx

```diff
@@ -84,9 +84,12 @@ export const EntityDetailsMultiformOverlay = ({
       planName: selectedEntity?.name,
     }) as EntityDetailsMultiformVerbiage;
 
-    const handleChildSubmit = (enteredData: AnyObject) => {
-      const updatedEntity = { ...selectedEntity, ...enteredData };
-      setSelectedEntity(updatedEntity);
+    const handleChildSubmit = (
+      enteredData: AnyObject,
+      updatedEntity?: AnyObject
+    ) => {
+      const entity = updatedEntity ?? selectedEntity;
+      setSelectedEntity({ ...entity, ...enteredData });
```

Comment:
if we deleted keys from the selected entity, we need to save that new trimmed data object (which sources from the `selectedEntity` at the start of its logic)